src/containers/page-layout/page-layout.tsx

```diff
@@ -0,0 +1,30 @@
+import { type FC } from "react";
+import { Stack, type StackProps } from "@mui/material";
+
+import { layout } from "~/constants/layout.js";
+
+import { PageHeader } from "../page-header/index.js";
+
+const styles = {
+  container: {
+    alignItems: "center",
+  },
+  content: {
+    width: layout.width.relative,
+    maxWidth: layout.width.max,
+    minWidth: layout.width.min,
+    alignItems: "center",
+  },
+};
+
+export const PageLayout: FC<{ title?: string } & StackProps> = ({
+  title,
+  children,
+}) => (
+  <Stack sx={styles.container}>
+    <Stack sx={styles.content}>
+      {title && <PageHeader text={title} />}
+      {children}
+    </Stack>
+  </Stack>
+);
```

Comment:
```suggestion
export const PageLayout: FC<{ title?: string } & StackProps> = ({
  title,
  children,
  stackProps,
}) => (
  <Stack sx={styles.container} {...stackProps}>
    <Stack sx={styles.content}>
      {title && <PageHeader text={title} />}
      {children}
    </Stack>
  </Stack>
);
```
since you are declaring it in the type let's be consistent

Reply:
This is intentional, I'm adding the `StackProps` type because I want to use the `children` prop but I do not want the layout taking any other prop from its parent. This will ensure every instance of <PageLayout /> serves the same styles to its children

Reply:
Then just declare children as ReactNode and don't extend the type since those other props will never be used and that may lead to misusing this component in the future